rotkehlchen/chain/polygon_pos/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,166 @@
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any, Final
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.constants import ZERO_ADDRESS
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.polygon.constants import CPT_POLYGON, CPT_POLYGON_DETAILS
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    ActionItem,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+WITHDRAW_TOPIC: Final = b'\xec:\xfb\x06{\xce3\xc5\xa2\x94G\x0e\xc5\xb2\x9egY0\x1c\xd3\x92\x85PI\x0cmH\x81l\xdc/]'  # noqa: E501
+DEPOSIT_FOUR_BYTES: Final = b'\x2e\x1a\x7d\x4d'
```

Comment:
https://polygonscan.com/tx/0xa97396d6b61f213ebf194807553c06768fb9d1ca04ba30bfafb1a788357f3c36

this 4 bytes is the one you use to decode this transaction right? ˆ  If so please don't use a name like deposit. I expected it to be `withdrawal`. If you mean `deposit to bridge` write `DEPOSIT_TO_BRIDGE_FOUR_BYTES` but is better to use the name of the function.

In this case I believe we have a problem because the 4 bytes detection is not unique. Any function with the same signature in polygon would hit that function. We need to improve the way we detect it as a bridge transaction :(

Reply:
Is this token something special? Other tokens don't have this `withdraw` method so I believe is not something standard. Do you know what it is? https://polygonscan.com/address/0xc5102fe9359fd9a28f877a67e36b0f050d81a3cc#writeContract

Reply:
Updated the names here to match the method they refer to. See question in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1300880132428398703) regarding the rest of the problem.

Reply:
Updated the filtering as discussed in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1300908720569454653)